composer.json

```diff
@@ -11,21 +11,38 @@
         }
     ],
     "require": {
+        "php": ">=5.3",
+        "pagerfanta/pagerfanta": "1.0.*",
         "symfony/framework-bundle": "~2.3|~3.0|~4.0",
-        "symfony/twig-bundle": "~2.3|~3.0|~4.0",
         "symfony/property-access": "~2.3|~3.0|~4.0",
-        "pagerfanta/pagerfanta": "1.0.*"
+        "symfony/twig-bundle": "~2.3|~3.0|~4.0"
     },
     "require-dev": {
         "symfony/symfony": "~2.3|~3.0|~4.0",
-        "phpunit/phpunit": "~3.7|~4.0"
+        "phpunit/phpunit": "~3.7|~4.0|^5.0"
     },
     "autoload": {
-        "psr-4": { "WhiteOctober\\PagerfantaBundle\\": "" }
+        "classmap": [
+            "WhiteOctoberPagerfantaBundle.php"
+        ],
+        "psr-4": {
+            "WhiteOctober\\PagerfantaBundle\\DependencyInjection\\": "DependencyInjection",
+            "WhiteOctober\\PagerfantaBundle\\EventListener\\": "EventListener",
+            "WhiteOctober\\PagerfantaBundle\\Twig\\": "Twig",
+            "WhiteOctober\\PagerfantaBundle\\View\\": "View"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "WhiteOctober\\PagerfantaBundle\\Tests\\": "Tests/"
+        }
     },
     "extra": {
         "branch-alias": {
             "dev-master": "1.0.x-dev"
         }
+    },
+    "config": {
+        "sort-packages": true
```

Comment:
`"optimize-autoloader": true,`, perhaps ?

Reply:
Sounds good, but when using the package in a project with `composer`, this flag is not followed right? Only if it is set in the projects `composer.json` (which includes this package) will it be done? I believe I was told so as a reason why this is not set @ Twig for example.

Reply:
if `optimize-autoloader` is put to package A, it will optimize autoloader for installation of deps when you are developing package A itself.
if you use package A as a requirement, `optimize-autoloader` or any part of `config` of package A, would be used

Reply:
https://getcomposer.org/doc/articles/autoloader-optimization.md says:
> Note: You should not enable any of these optimizations in development as they all will cause various problems when adding/removing classes. The performance gains are not worth the trouble in a development setting.

However, adding `"optimize-autoloader": true,` here would optimise in both development and production, so we wouldn't want that.

Given that, I'm happy to leave this out for now - this feels best solved by deployment scripts calling `install` or `update` with `--optimize-autoloader`, which is beyond the scope of what we can control in this library!

Thanks for the suggestion, though - I learned something today!

Reply:
never experienced any troubles of that kind ;)